### ISSUE TITLE
[YSQL] Mis-numbered days of the week in example

### ISSUE THREAD

**Issue by GeekFiftyFive** [yugabyte/yugabyte-db#11807]
Jira Link: [DB-910](https://yugabyte.atlassian.net/browse/DB-910)
### Description

In the documentation for `array_position` and `array_positions`, found here https://docs.yugabyte.com/latest/api/ysql/datatypes/type_array/functions-operators/array-position/#array-position, it appears that the days of the week are mis-numbered in the given example.

The numbering jumps from 7 to 9, which also happens to cause inconsistency with the result of running the example query (i.e. searching for `'mon'` starting with index `3` would give a value of `8`, but the numbering in the example data would be `9`).

**Comment by bllewell** [comment]
It took some time before this issue was assigned to me. (Yes, I wrote the Arrays doc.) It's good to know that a user has read it carefully. It seems that I did a silly typo all those years ago. Here's a better example that uses _generate_subscripts()_ after defining the example array to show what element is at what index:

```
create or replace view v(days_arr) as
select array[
              'mon', 'tue', 'wed', 'thu','fri', 'sat', 'sun', 
              'mon', 'tue', 'wed', 'thu','fri', 'sat', 'sun',
              'mon', 'tue', 'wed'
            ]::text[];

with c(days, idx) as (
  select v.days_arr, subscripts.idx
  from v
  cross join lateral
  generate_subscripts(v.days_arr, 1) as subscripts(idx))
select idx, days[idx] as day from c order by idx;
```

This is the result:

```
 idx | day 
-----+-----
   1 | mon
   2 | tue
   3 | wed
   4 | thu
   5 | fri
   6 | sat
   7 | sun
   8 | mon
   9 | tue
  10 | wed
  11 | thu
  12 | fri
  13 | sat
  14 | sun
  15 | mon
  16 | tue
  17 | wed
```

**array_position() example**

```
select array_position(
    (select days_arr from v),  -- The to-be-searched array.
    'tue'::text,               -- The to-be-searched-for value.
    3::int                     -- The (inclusive) position
                               --   at which to start searching. [optional]
  ) as position;
```

This is the result

```
 position 
----------
        9
```

**array_positions() example**

```
select unnest(
  array_positions(
      (select days_arr from v),  -- The to-be-searched array.
      'tue'::text                -- The to-be-searched-for value.
    )
  ) as position;
```

This is the result

```
 position 
----------
        2
        9
       16
```

**Comment by bllewell** [comment]
Fixed by [PR #17201](https://github.com/yugabyte/yugabyte-db/pull/17201), Commit [d947db5](https://github.com/yugabyte/yugabyte-db/pull/17201/commits/d947db5077750e6cb5f37af0fcb9535c90f0a65f)
